### DISCUSSION TITLE
Kyuubi use SPARK_SQL as an engine, and it is very slow to get metadata information using a jdbc connection

### DISCUSSION THREAD

**Discussion by as10128** [apache/kyuubi#4956] [Q&A]
Kyuubi use SPARK_SQL as an engine, and it is very slow to get metadata information (Getschemas Gettables Getcolumns) using a jdbc connection, and Kyuubi logs show 1 minute. But using beeline is fast, how should I check this issue ?
<img width="1648" alt="image" src="https://github.com/apache/kyuubi/assets/22188671/89895130-69fd-41de-b91b-b55e3bb44e21">
<img width="1648" alt="image" src="https://github.com/apache/kyuubi/assets/22188671/d3a46b8b-c687-4caf-b357-500b2882c7a8">

**Comment by pan3793** [comment]
The performance of `Getschemas` `Gettables` `Getcolumns` has room to improve, e.g. parallel to get those information.

And there is a workaround `kyuubi.operation.getTables.ignoreTableProperties=true` to skip fetching some information to speed up the method.

**Reply by as10128** [reply]
@pan3793 Thanks, but I still have a question, Getschemas Gettables Getcolumns takes about the same time, is it every time you get metadata to re-fetch all the information and then filter?

**Reply by pan3793** [reply]
Almost, since there is no pushdown support.

**Comment by pan3793** [comment]
FYI: `kyuubi.operation.getTables.ignoreTableProperties=true` was enhanced to support session catalog cases in https://github.com/apache/kyuubi/pull/6018
